src/sparse_ci/ui64_determinant.cc

```diff
@@ -287,6 +287,70 @@ double UI64Determinant::destroy_beta_bit(int n) {
     return slater_sign_b(n);
 }
 
+std::vector<std::vector<int>> UI64Determinant::get_asym_occ(int norb, std::vector<int> act_mo) const { 
+
+    size_t nirrep = act_mo.size();
+    std::vector<std::vector<int>> occ(nirrep);
+
+    int abs = 0;
+    for( int h = 0; h < nirrep; ++h ){
+        for( int p = 0; p < act_mo[h]; ++p ){
+            if (get_alfa_bit(abs)) {
+                occ[h].push_back(abs);
+            }
+            abs++;
+        }
+    }
+    return occ;
+}
+
+std::vector<std::vector<int>> UI64Determinant::get_bsym_occ(int norb, std::vector<int> act_mo) const { 
+    size_t nirrep = act_mo.size();
+    std::vector<std::vector<int>> occ(nirrep);
+
+    int abs = 0;
+    for( int h = 0; h < nirrep; ++h ){
+        for( int p = 0; p < act_mo[h]; ++p ){
+            if (get_beta_bit(abs)) {
+                occ[h].push_back(abs);
+            }
+            abs++;
+        }
+    }
+    return occ;
+}
+
+std::vector<std::vector<int>> UI64Determinant::get_asym_vir(int norb, std::vector<int> act_mo) const { 
+    size_t nirrep = act_mo.size();
+    std::vector<std::vector<int>> occ(nirrep);
+
+    int abs = 0;
+    for( int h = 0; h < nirrep; ++h ){
+        for( int p = 0; p < act_mo[h]; ++p ){
+            if (not get_alfa_bit(abs)) {
+                occ[h].push_back(abs);
+            }
+            abs++;
+        }
+    }
+    return occ;
+}
+
+std::vector<std::vector<int>> UI64Determinant::get_bsym_vir(int norb, std::vector<int> act_mo) const { 
+    size_t nirrep = act_mo.size();
+    std::vector<std::vector<int>> occ(nirrep);
+
+    int abs = 0;
+    for( int h = 0; h < nirrep; ++h ){
+        for( int p = 0; p < act_mo[h]; ++p ){
+            if (not get_beta_bit(abs)) {
+                occ[h].push_back(abs);
+            }
+            abs++;
+        }
+    }
+    return occ;
+}
```

Comment:
All of these look very inefficient due to allocation of multiple vectors. Are they called for each determinant? It might make sense to pass a `std::vector<std::vector<int>>` by reference and fill it up.

Reply:
Yeah, this is called for every reference determinant. I'll work on a better implementation of these functions (norb isn't even used?). Passing the object pre-allocated does make sense.